lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -410,8 +428,11 @@ def authorization_from(raw)
       end
 
       def credit_fund_transfer_attribute(options)
-        return unless options[:credit]
-        {'action' => "REFUND"}
+        if options[:credit]
+          {'action' => 'REFUND'}
+        elsif options[:action]
+          {'action' => options[:action]}
+        end
```

Comment:
The `elsif` in this context caught my eye. Did you mean `else`? This is a great spot for a ternary operator. 

`options[:credit] ? { 'action' => 'REFUND' } : { 'action' => options[:action] }`

Also, what will happen if there is no `options[:action]`?

Reply:
I did mean `elsif`, not `else`. In the case neither branch is taken, `nil` is returned, and XML attributes will be added to the `paymentDetails` tag, which is indeed what we want as the default. This is also why I didn't go with a ternary; you'd have to use two of them to achieve the effect, and doing something like

    return if option[:action] && options[:action] != :credit
    {'action' => options[:action] == :credit ? 'REFUND' : options[:action]}

felt more confusing to me than the version I did. I'm open to alternatives, though; this conditional definitely feels odd, even if it's mechanically correct, and I did feel as if there was probably a cleaner approach.

Reply:
@deedeelavinder Actually, after sleeping on it, I think the second is actually better, because it's more obvious that the code isn't making a mistake and does intend to return `nil` in that use case. I'll swap that out. Thanks!

Reply:
Landed. Slightly different than the above code-snippet, which is wrong, but same idea.

Reply:
@bpollack You could also set a variable for use in the ternary (so that `nil` is returned when appropriate), but the latest also looks good.